snuba/query/processors/physical/null_column_caster.py

```diff
@@ -1,13 +1,15 @@
-from typing import Dict, Sequence
+from typing import TYPE_CHECKING, Dict, Sequence
 
 from snuba.clickhouse.columns import FlattenedColumn, SchemaModifiers
 from snuba.clickhouse.query import Query
-from snuba.datasets.storage import ReadableTableStorage
 from snuba.query.expressions import Column, Expression, FunctionCall, Literal
 from snuba.query.functions import AGGREGATION_FUNCTIONS
 from snuba.query.processors.physical import ClickhouseQueryProcessor
 from snuba.query.query_settings import QuerySettings
 
+if TYPE_CHECKING:
```

Comment:
What happened here?

Reply:
This is done to prevent a circular import. ReadableTableStorabe is only necessary for type checking but importing it during runtime causes an import cycle. I don't remember the cycle path exactly but this is a documented way to solve import cycle issues in mypy

https://mypy.readthedocs.io/en/stable/runtime_troubles.html#typing-type-checking

Reply:
Sorry I should have been more clear, what was the circular import, and was that import caused by our code being poorly laid out?

Reply:
`snuba.datasets.storage` import imports ClickhouseQueryProcessor (for type checking)
this file imports `ReadableTableStorage` for type checking

Reply:
Got it, OK thanks.